TweetMate/src/main/java/com/ybsystem/tweetmate/fragments/fragment/EasyTweetFragment.java

```diff
@@ -0,0 +1,213 @@
+            if (hasFocus) {
+                // If text is empty, then set HashTag
+                if (mPostEdit.getText().length() == 0 && hasHashTag()) {
+                    mPostEdit.setText("\n" + mSearchWord);
+                    new Handler().postDelayed(() -> mPostEdit.setSelection(0), 0);
+                }
+            }
+        });
+
+        // When text changed
+        mPostEdit.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                int length = s.toString().length();
+                int color = length > 140 ? Color.RED : COLOR_TEXT;
+
+                // Change count
+                mTextCount.setText(Integer.toString(140 - length));
+                mTextCount.setTextColor(color);
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+            }
+
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+            }
+        });
+    }
+
+    private void setPostButton(View view) {
+        // When clicked
+        view.findViewById(R.id.button_post).setOnClickListener(v -> {
+            // Check text length
+            int length = mPostEdit.getText().length();
+            if (length == 0) {
+                intentToPostActivity(v.getContext());
+                return;
+            }
+            // Create tweet and post
+            StatusUpdate update = new StatusUpdate(
+                    mPostEdit.getText().toString()
+            );
+            StatusUseCase.post(update, new ArrayList<>());
+
+            // Save hashtag
+            EntityArray<String> hashtags = TweetMateApp.getMyAccount().getHashtags();
+            String regex = "[#＃](w*[一-龠_ぁ-ん_ァ-ヴーａ-ｚＡ-Ｚa-zA-Z0-9]+|[a-zA-Z0-9_]+|[a-zA-Z0-9_]w*)";
+            Matcher matcher = Pattern.compile(regex).matcher(mPostEdit.getText());
+            while (matcher.find()) {
+                if (hashtags.size() >= 5) {
+                    hashtags.remove(0);
+                }
+                hashtags.add(matcher.group());
+            }
+        });
+
+        // When long clicked
+        view.findViewById(R.id.button_post).setOnLongClickListener(v -> {
+            intentToPostActivity(v.getContext());
+            return true;
+        });
+    }
+
+    private void intentToPostActivity(Context context) {
+        Intent intent = new Intent(context, PostActivity.class);
+        if (hasHashTag()) intent.putExtra("TWEET_SUFFIX", mSearchWord);
+        startActivityForResult(intent, 0);
+        getActivity().overridePendingTransition(R.anim.fade_in_from_bottom, R.anim.none);
+    }
+
+    private boolean hasHashTag() {
+        // Check word
+        if (mSearchWord == null) {
+            return false;
+        }
+        if (!mSearchWord.startsWith("#") && !mSearchWord.startsWith("＃")) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AS1126)